src/core/bsequence.ml

```diff
@@ -0,0 +1,49 @@
+
+type +'a node =
+    | Nil
+    | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
```

Comment:
this is a good point. While I understand wanting to avoid a lot of the category theory nomenclature, it might be good to keep the basics to it's clear what the structures being used actually are? `return`, `pure` or something like that are pretty standard in FP.